src/object_store.cpp

```diff
@@ -74,6 +79,22 @@ void create_metadata_tables(Group& group) {
         pk_table->insert_column(c_primaryKeyPropertyNameColumnIndex, type_String, c_primaryKeyPropertyNameColumnName);
     }
     pk_table->add_search_index(c_primaryKeyObjectClassColumnIndex);
+
+#if REALM_ENABLE_SYNC
+    // Only add __ResultSets if Realm is a partial Realm
+    if (partial_realm && !group.has_table(result_sets_type_name)) {
```

Comment:
Can you explain the motivation behind performing this work here, rather than doing it when the query is registered where it's still necessary to make schema changes too (adding the matches column)?

Reply:
The code doesn't need to be moved; it can just be deleted since there's already code for doing this.